pkg/reconciler/route/route.go

```diff
@@ -365,17 +365,22 @@ func (c *Reconciler) tls(ctx context.Context, host string, r *v1alpha1.Route, tr
 func (c *Reconciler) reconcileDeletion(ctx context.Context, r *v1alpha1.Route) error {
 	logger := logging.FromContext(ctx)
 
+	// Delete the Ingress resources for this Route.
+	logger.Info("Cleaning up Ingress")
```

Comment:
The loop below is actually incorrect.  Please revert this to what it was last release: https://github.com/knative/serving/blob/b066e90ccd6f091e0faf65a759f3ce789d373457/pkg/reconciler/route/route.go#L377-L397

Finalizers should only be evaluated when they are at the head of the list.

Reply:
> Finalizers should only be evaluated when they are at the head of the list.

I understand the principle - but do you have more context? I don't see this recommendation anywhere (ie. kubebuilder book/k8s docs).

Reply:
I don't think there's requirement for that, but it might be the notion we did for ourselves (i.e. we are the last ones who reference the object).

Reply:
Sure, reverted it now.